cropper/src/main/java/com/canhub/cropper/CropImageActivity.kt

```diff
@@ -323,4 +349,168 @@ open class CropImageActivity :
+        context: Context,
+        title: CharSequence?,
+        includeCamera: Boolean,
+        includeGallery: Boolean,
+        includeDocuments: Boolean
+    ): Intent {
+        val allIntents: MutableList<Intent> = ArrayList()
+        val packageManager = context.packageManager
+        // collect all camera intents if Camera permission is available
+        if (!isExplicitCameraPermissionRequired(context) && includeCamera) {
+            allIntents.addAll(getCameraIntents(context, packageManager))
+        }
+        if (includeGallery) {
+            var galleryIntents =
+                getGalleryIntents(packageManager, Intent.ACTION_GET_CONTENT, includeDocuments)
+            if (galleryIntents.isEmpty()) {
+                // if no intents found for get-content try pick intent action (Huawei P9).
+                galleryIntents =
+                    getGalleryIntents(packageManager, Intent.ACTION_PICK, includeDocuments)
+            }
+            allIntents.addAll(galleryIntents)
+        }
+        val target: Intent
+        if (allIntents.isEmpty()) {
+            target = Intent()
+        } else {
+            target = Intent(Intent.ACTION_CHOOSER, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
+            if (includeGallery) {
+                target.action = Intent.ACTION_PICK
+                target.type = "image/*"
+            }
+        }
+        // Create a chooser from the main  intent
+        val chooserIntent = Intent.createChooser(target, title)
+        // Add all other intents
+        chooserIntent.putExtra(
+            Intent.EXTRA_INITIAL_INTENTS, allIntents.toTypedArray<Parcelable>()
+        )
+        return chooserIntent
+    }
+
+    /**
+     * Get all Camera intents for capturing image using device camera apps.
+     */
+    private fun getCameraIntents(context: Context, packageManager: PackageManager): List<Intent> {
+        val allIntents: MutableList<Intent> = ArrayList()
+        // Determine Uri of camera image to  save.
+        val outputFileUri = getTmpFileUri()
+        latestTmpUri = outputFileUri
+        val captureIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
+        val listCam = packageManager.queryIntentActivities(captureIntent, 0)
+        for (res in listCam) {
+            val intent = Intent(captureIntent)
+            intent.component = ComponentName(res.activityInfo.packageName, res.activityInfo.name)
+            intent.setPackage(res.activityInfo.packageName)
+            if (context is Activity) {
+                context.grantUriPermission(
+                    res.activityInfo.packageName, outputFileUri,
+                    Intent.FLAG_GRANT_WRITE_URI_PERMISSION or Intent.FLAG_GRANT_READ_URI_PERMISSION
+                )
+            }
+            intent.putExtra(MediaStore.EXTRA_OUTPUT, outputFileUri)
+            allIntents.add(intent)
+        }
+        return allIntents
+    }
+
+    private val famousGalleryPackages = listOf(
```

Comment:
Do we have a link for where we can check this list and update?

Reply:
No. I've actually added this at the last minute. 

What it does is just check and add some of the most common apps to the start of the list instead of the end. This makes it easier for users to pick images from more well known apps. A few I could think of are Google Photos, Google Photos Go, Samsung Gallery, MIUI Gallery. These are the ones I can confirm are the right apps since I could test them.

Reply:
We should make this list configurable for the users

Reply:
Alright I'll do this